modules/heat_transfer/src/materials/ElectromagneticHeatingMaterial.C

```diff
@@ -0,0 +1,110 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElectromagneticHeatingMaterial.h"
+#include "FormulationEnums.h"
+
+registerMooseObject("HeatTransferApp", ElectromagneticHeatingMaterial);
+
+InputParameters
+ElectromagneticHeatingMaterial::validParams()
+{
+  InputParameters params = ADMaterial::validParams();
+  params.addClassDescription(
+      "Material class used to provide the electric field as a material property and computes the "
+      "residuals for electromagnetic/electrostatic heating based objects.");
+  params.addCoupledVar(
+      "field", "The electic field vector or electrostatic potential scalar to produce the field.");
+  params.addCoupledVar(
+      "complex_field",
+      "The complex component of the electic field vector for the harmonic formulation.");
+  params.addParam<std::string>(
+      "field_material_name", "field", "User-specified material property name for the field.");
+  params.addParam<std::string>("field_heating_name",
+                               "field_heating",
+                               "User-specified material property name for the Joule Heating.");
+  params.addParam<Real>("heating_scaling", 1.0, "Coefficient to multiply by heating term.");
+  params.addParam<MaterialPropertyName>(
+      "conductivity",
+      "conductivity",
+      "Material property providing electrical conductivity of the material.");
+  MooseEnum formulation("time frequency", "time");
+  MooseEnum solver("electrostatic electromagnetic", "electrostatic");
+  params.addParam<MooseEnum>(
+      "formulation",
+      formulation,
+      "The domain formulation of the Joule heating, time or frequency (default = time).");
+  params.addParam<MooseEnum>(
+      "solver", solver, "Electrostatic or electromagnetic field solver (default = electrostatic).");
+  return params;
+}
+
+ElectromagneticHeatingMaterial::ElectromagneticHeatingMaterial(const InputParameters & parameters)
+  : ADMaterial(parameters),
+    _field_var(*getFieldVar("field", 0)),
+    _is_vector(_field_var.isVector()),
+    _efield(_is_vector ? adCoupledVectorValue("field") : _ad_grad_zero),
+    _efield_complex(_is_vector ? adCoupledVectorValue("complex_field") : _ad_grad_zero),
+    _grad_potential(_is_vector ? _ad_grad_zero : adCoupledGradient("field")),
+    _field(declareADProperty<RealVectorValue>(getParam<std::string>("field_material_name"))),
+    _field_complex(declareADProperty<RealVectorValue>(getParam<std::string>("field_material_name") +
```

Comment:
similar to the conductivity, I'd recommend a name for these variables that includes `electric`: my default assumption for a field in the heat transfer module is a thermal field

Reply:
Similarly to previous comment, I changed it to `electric_field`.